questions.md

```diff
@@ -41,8 +53,12 @@ It's my first project with Solana / blockchain so I have several questions for m
     },
     ```
 
+Yes, it's ok. When a transaction is serialized, account keys are grouped by signer / writable properties. But instructions can use those accounts in any order (and can use the same account multiple times).
+
 6. How to properly create a `Vec<Party>` on chain? (The current implementation creates party accounts with PDAs as pubkeys. One seed used to generate these PDAs is set to the next index value. The current index is set in a "shared" account and incremented in the create account transaction. It's a bit ugly and there may be conflicts because the current index value is read outside of the transaction.)
 
+If your program essentially has a global shared list of ordered accounts, then you're going to have contention for the next account. You could lessen the impact though by having each transaction include the next X party accounts and iterate through to find one that isn't initialized yet? But either way, your program can get DOS'd by someone who tries to front run whenever a new account is made. Another approach is a reservation system where you reserve the next index for a requesting user and they can fulfill that reservation async. You would have some holes in the list though.
+
```

Comment:
I was thinking about other approaches (all of them have different advantages and disadvantages of course):

1. Store all Party pubkeys in the parent account, inside an array with a fixed length.
2. Create a linked list from Party accounts and store the head and the last account in the parent account. The backend would cache accounts to avoid doing too many requests while iterating accounts.
3. Don't store anything in the parent account at all. Call JSON RPC [getProgramAccounts](https://docs.solana.com/developing/clients/jsonrpc-api#getprogramaccounts) and filter the accounts by the parent account pubkey stored in the iterated account data. However `getProgramAccounts` doesn't support pagination (?) and there is probably a hardcoded result limit (?).

Reply:
Option 3 is probably the way to go once generic indexing services exist (they are coming).